src/controllers/controller.bar.js

```diff
@@ -210,19 +218,23 @@ module.exports = DatasetController.extend({
 		var me = this;
 		var chart = me.chart;
 		var scale = me._getIndexScale();
+		var metasets = chart._getSortedVisibleDatasetMetas();
 		var stacked = scale.options.stacked;
-		var ilen = last === undefined ? chart.data.datasets.length : last + 1;
+		var ilen = metasets.length;
 		var stacks = [];
 		var i, meta;
 
 		for (i = 0; i < ilen; ++i) {
-			meta = chart.getDatasetMeta(i);
-			if (meta.bar && chart.isDatasetVisible(i) &&
+			meta = metasets[i];
+			if (meta.bar &&
```

Comment:
This might be able to be simplified to fix another Code Climate warning. Would it be equivalent to write this?
```
if (meta.bar && (stacked === false || stacks.indexOf(meta.stack) < 0)) {
```